### PR TITLE
(Re-)Introduce Typescript to griff-react (#263) (#274)

### PR DESCRIPTION
**Pull request by evancharlton**
This reverts commit 6e0bc4f9ad5f5222136f45879c02692154982076.

Additionally, this migrates from nwb to rollup so that .js files get
bundled along with the .ts{,x} files during the migration.

Additionally, this removes the demo since Storybook is a *much* better
demo arena.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-278.surge.sh

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-278.surge.sh

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-278.surge.sh

File: Jenkinsfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ def label = "${REPO}-${UUID.randomUUID().toString().substring(0, 5)}"
 podTemplate(
   label: label,
   containers: [containerTemplate(name: 'node',
-                                image: 'node:9',
+                                image: 'node:8',
```
Comment (valerii-cognite) [inline Jenkinsfile]: `node:10` won't work?

File: nwb.config.js [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
-  }
-}
+    umd: false,
+  },
+  webpack: {
+    extra: {
+      entry: path.resolve('./build/src'),
+    },
+  },
+};
```
Comment (valerii-cognite) [inline nwb.config.js]: You probably don't need this configuration anymore

File: package.json [hunk 3]
```diff
@@ -11,19 +11,20 @@
...
-    "build:watch": "nodemon -w src -x 'rm .babelrc &> /dev/null; nwb build-react-component --no-demo'",
-    "test:coverage": "nwb test-react --coverage",
-    "test:watch": "nwb test-react --server",
-    "storybook": "start-storybook -p 9001 -c .storybook",
-    "storybook:build": "build-storybook -c .storybook -o .out",
-    "demo:build": "nwb build-react-component src/index.js",
+    "build": "yarn clean && yarn compile && rollup -c",
+    "build:watch": "yarn rollup -w -c",
+    "clean": "rm -rf build/ es/ lib/",
+    "compile": "tsc -p .",
```
Comment (valerii-cognite) [inline package.json]: I'm not sure that we need this step with rollup

File: tslint.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "no-default-export": false,
+    "variable-name": [
+      true,
+      "ban-keywords",
+      "check-format",
+      "allow-pascal-case"
+    ],
+    "max-line-length": [true, 100]
+  }
+}
```
Comment (valerii-cognite) [inline tslint.json]: tslint or gts?

File: Jenkinsfile (hunk 1 above)
Comment (evancharlton) [reply Jenkinsfile]: dunno; lemme try.

File: nwb.config.js (hunk 2 above)
Comment (evancharlton) [reply nwb.config.js]: Good catch!

File: package.json (hunk 3 above)
Comment (evancharlton) [reply package.json]: We do to get the files into `build/` -- but I'll see if we even need that anymore.

File: tslint.json (hunk 4 above)
Comment (evancharlton) [reply tslint.json]: `gts` uses `tslint` under the hood.

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-278.surge.sh

File: Jenkinsfile (hunk 1 above)
Comment (evancharlton) [reply Jenkinsfile]: Seems to work.

File: package.json (hunk 3 above)
Comment (evancharlton) [reply package.json]: I'm sure there's a way to configure storybook to use `babel-loader` (or something?) to load the `.ts` files directly, but I haven't figured that out yet. I'm going to stick with the `.ts + .js => build/**/*.js` configuration until there's a better solution.

**Pull request merged**